st2reactor/st2reactor/garbage_collector/base.py

```diff
@@ -0,0 +1,158 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Garbage collection service which deletes old data from the database.
+"""
+
+import datetime
+
+import eventlet
+from eventlet.support import greenlets as greenlet
+from oslo_config import cfg
+
+from st2common import log as logging
+from st2common.constants.exit_codes import SUCCESS_EXIT_CODE
+from st2common.constants.exit_codes import FAILURE_EXIT_CODE
+from st2common.util import isotime
+from st2common.util.date import get_datetime_utc_now
+from st2reactor.garbage_collector.executions import purge_executions
+from st2reactor.garbage_collector.trigger_instances import purge_trigger_instances
+
+__all__ = [
+    'GarbageCollectorService'
+]
+
+# TODO: Make config option
+DEFAULT_COLLECT_INTERVAL = 300
+
+# Minimum value for TTL. If user supplies value lower than this, we will throw.
+MINIMUM_TTL_DAYS = 7
```

Comment:
I think it's good to have safe-guard, but I could be convinced to lower this threshold.

In theory, we could also support very low TTLs (less than one day), but specifying TTL that low indicates a problem elsewhere and would cause all kinds of issues (MongoDB would probably have tons of issues catching up with the compactions and performance would suck - our current data and storage model is not optimized for that).